graph/kosaraju.go

```diff
@@ -0,0 +1,61 @@
+// Kosaraju Algorithm is used to get strongly connected components of a graph
+// Time Complexity: O(N + M), where N is number of vertices and M is number of edges
+// Space Complexity: O(N)
+// Reference: https://www.baeldung.com/cs/kosaraju-algorithm-scc
+
+package graph
+
+func topologicalSort(node int, adj [][]int, sortedNodes *[]int, visited map[int]bool) {
+	visited[node] = true
+	for _, nbr := range adj[node] {
+		if visited[nbr] {
+			continue
+		}
+		topologicalSort(nbr, adj, sortedNodes, visited)
+	}
+	*sortedNodes = append(*sortedNodes, node)
+}
+
+func getTransposeGraph(n int, adj [][]int) [][]int {
+	transposeGraph := make([][]int, n+1)
+	for i := 1; i <= n; i++ {
+		for _, nbr := range adj[i] {
+			transposeGraph[nbr] = append(transposeGraph[nbr], i)
+		}
+	}
+	return transposeGraph
+}
+
+func dfs(node int, adj [][]int, visited map[int]bool, currentSCC *[]int) {
+	visited[node] = true
+	*currentSCC = append(*currentSCC, node)
+	for _, nbr := range adj[node] {
+		if visited[nbr] {
+			continue
+		}
+		dfs(nbr, adj, visited, currentSCC)
+	}
+}
+
+func GetStronglyConnectedComponents(n int, adj [][]int) [][]int {
```

Comment:
I think it would have been much more easier to understand this algorithm if you had used the `Graph` data structure defined in the `graph.go` file. There are already a lot of helpers for that graph as well.

Reply:
If you don't need the weighted edges, then don't use the weighted edges part of the graph 😄.